src/ec-evaluator/utils.ts

```diff
@@ -199,6 +199,29 @@ export const createBlockEnvironment = (
   }
 }
 
+/**
```

Comment:
Wouldn't it be simpler to avoid creating empty frames in the first place? An environment extension should just take the given environment if there are no bindings to be added.